### ISSUE TITLE
Naming convention issue - moved from metrics repo

### ISSUE THREAD

**Issue by klumb** [chaoss/wg-gmd#18]
I am going through the metrics markdown files and finding a lot of repetition. I would like to push a fix that will allow us to get these pages up to date and easier to work with.

Besides some basic editing, I think we need to agree on a standard naming convention for these files. I propose:
"the name of the resource being measured"-"more_description_if_necessary"- "the_measurement_if_necessary".md

Ex:
issues-open-average_time_distribution.md

This example metric describes the average number of issues over a period of time.

By doing this we can easily sort and identify related metrics.

Thoughts?

**Comment by ccarterlandis** [comment]
I think this is very much needed. In working on the Augur metrics status page, I realized two concepts: an `ID` and a `tag`. While it applies to Augur in a slightly different way, I think that it could still be useful.

Within Augur's idea of metrics, each metric has a `tag`, which is what the working group defines as the metric's "name" if you will. Each metric also has an `ID`, which is created by combining the metric's data source and it's tag. So if the `open-issues` metric (that's the `tag`) has an implementation using GHTorrent, the `ID` is `ghtorrent-open-issues`. This allows a metric to be defined in multiple data sources.

In the context of the working group, obviously the data source wouldn't apply, but perhaps something else could? I'd definitely like some standard way to identify (and thus classify) metrics. It sounds like what you're proposing is rather similar anyways - just some food for thought!

**Comment by klumb** [comment]
Yes @GeorgLink

Sorry I should have linked it.

**Comment by klumb** [comment]
Please see https://github.com/klumb/wg-gmd
for an example.

Please note that we are talking about the naming convention for the files in the resource folder and not the content of the file. I believe the files will need to be edited as well but that is a different issue.

**Comment by GeorgLink** [comment]
@ccarterlandis wrote:
> So if the `open-issues` metric (that's the `tag`) has an implementation using GHTorrent, the ID is `ghtorrent-open-issues`. This allows a metric to be defined in multiple data sources.
> ...
> In the context of the working group, obviously the data source wouldn't apply,

I see where you are coming from and it makes for an implementation like Augur. I agree also, that from the CHAOSS metrics perspective, we want to stay implementation agnostic and not tie a metric definition to the data source. What you call `tag` could then be the canonical definition and within that page, we could have a section talking about the intricacies of implementing it in different technologies. This would provide a nice overview of implementation challenges of the same metric across technologies.

**Comment by klumb** [comment]
Thanks for the google document @GeorgLink. I think D&Is Question/resource pages match up to the aggregate metrics pages we are wanting to create in GMD. 

I think one of the reasons there is confusion here is because we are coming at it from a different directions. We are working with activity metrics that are for the most part already defined. I believe our goal right now is to establish standards around these metric detail pages then we can begin to construct the question/resource/aggregate pages. 

This issue is about the standards for the metrics detail pages. Higher level discussions will follow.

**Comment by GeorgLink** [comment]
Perfect. Much to learn from this discussion.

**Comment by jgbarah** [comment]
Oooops. I'm sorry, this was below my radar.

Yes, I find the naming conventions important, mainly to use the same name to the same concept, and to have a convention that allows us to group names according to prefixes, etc. But I think this is discussion is different from how to do goal-question-metric, so I would have it in a separate ticket. And for me, using chaoss/wg-diversity-inclusion#66 for the second is perfect, so maybe we can just follow that discussion there, and then translate the results here, if convenient.

WRT to naming, with the current state of affairs, I would stop for now, after #24 was accepted, and close this issue (at least form now). @GeorgLink @ccarterlandis @klumb do you agree?

**Comment by klumb** [comment]
Yes, I believe we can close this.

**Comment by jgbarah** [comment]
OK, thanks. Closing...
